daemon/algod/api/server/v2/handlers.go

```diff
@@ -182,6 +182,64 @@ func (v2 *Handlers) GetBlock(ctx echo.Context, round uint64, params generated.Ge
 	return ctx.Blob(http.StatusOK, contentType, data)
 }
 
+// GetProof generates a Merkle proof for a transaction in a block.
+// (GET /v2/blocks/{round}/transactions/{txid}/proof)
+func (v2 *Handlers) GetProof(ctx echo.Context, round uint64, txid string, params generated.GetProofParams) error {
+	var txID transactions.Txid
+	err := txID.UnmarshalText([]byte(txid))
+	if err != nil {
+		return badRequest(ctx, err, errNoTxnSpecified, v2.Log)
+	}
+
+	ledger := v2.Node.Ledger()
+	block, _, err := ledger.BlockCert(basics.Round(round))
+	if err != nil {
+		return internalError(ctx, err, errFailedLookingUpLedger, v2.Log)
+	}
+
+	proto := config.Consensus[block.CurrentProtocol]
+	if proto.PaysetCommit != config.PaysetCommitMerkle {
+		return internalError(ctx, err, "protocol does not support Merkle proofs", v2.Log)
```

Comment:
I think this should be 412 rather than 5XX since there is no issue with the server; the network is currently at "fault" here and re-trying will not be helpful. One notation is that it's a "deterministic" error : trying making the request again would not help.

@rotemh - what do you think should the error code be in this case ?

Reply:
I changed it to 404.  Logically, I think you could make the argument that 404 makes some sense: this object does not exist, and will not exist, if the protocol doesn't support proofs at that block number.  412 seems to be relatively specific to `If-Modified-Since` and `If-None-Match`, which is not related to the error condition we're considering here.  But if @algorotem has a convincing argument for something else, I'm happy to change it.